app/volume.go

```diff
@@ -180,25 +180,29 @@ func (job *Job) snapshotAndCleanup() error {
 	}
 	snapshots, err := job.engine.SnapshotList()
 	if err != nil {
-		return err
+		logrus.Warnf("error getting list of snapshots for %v: %v", job.volumeName, err)
```

Comment:
Instead of modifying every return, we can use `defer func(){}` at the beginning of the function to deal with this if we don't want to error out.

see https://github.com/longhorn/longhorn-manager/blob/master/controller/volume_controller.go#L210

Reply:
There would be still a few places that we'd want to error out though, like on an actual `Backup` or on a `Snapshot` failure, right (So that the `Recurring Job` knows to attempt a `Backup` or a `Snapshot` multiple times)? In that case, wouldn't it be best to replace the return statements and then leave the existing `return err` lines in the places that we still want the `Job` to error out?

Reply:
Those operations will be performed at the beginning of the function? Then we can put `defer fund(){}` after those functions.